### DISCUSSION TITLE
What is the refresh rate?

### DISCUSSION THREAD

**Discussion by bkqc** [widewing/ha-toyota-na#96] [Q&A]
I just installed and configured the integeration on my setup and I'm amazed by the number  of sensors available! My concern is that they seem to update very sparsely. For instance, I left the trunk open for like 2 minutes without getting the value to change. I did a second test unlocking the doors and it took less than 4 minutes before the update arrived.

Are the values read through polling or received from pushing? If it is polling, at what frequency? Is it configurable?

**Comment by seang96** [comment]
https://github.com/widewing/ha-toyota-na/blob/master/custom_components%2Ftoyota_na%2Fconst.py#L20-L21

UPDATE_INTERVAL just looks up the latest data from Toyota, 30 seconds, this is like opening the app to see the statues. This data may be old if the car hasn't sent data recently.

REFRESH_STATUS_INTERVAL, 60mins, sends a request to the car to get its current info. Getting data from the car has a delay since they have to "wake" the car up. It probably also uses the battery of your car slightly more. This functions as the "refresh" button in the app. The car also updates by itself after a period of time if there are certain actions done to it like unlocking it.

You can manually refresh the car using the refresh action in your automations, this does the same thing as the 60min one.
